samples/sample_decode/src/pipeline_decode.cpp

```diff
@@ -1566,12 +1595,21 @@ mfxStatus CDecodingPipeline::DeliverOutput(mfxFrameSurface1* frame)
 
     if (m_bExternalAlloc) {
         if (m_eWorkMode == MODE_FILE_DUMP) {
-            res = m_pGeneralAllocator->Lock(m_pGeneralAllocator->pthis, frame->Data.MemId, &(frame->Data));
```

Comment:
Shouldn't we execute this line in case of VACopy?

Reply:
No, we needn't. This called function will execute vaMapBuffer and vaDeriveImage to map data between VA surface and user buffer. But VACopy doesn't need vaMapBuffer or vaDeriveImage. It copies data between two surfaces directly.